### ISSUE TITLE
broken document link - Sylvilagus floridanus (J. A. Allen, 1890) https://www.mammaldiversity.org/taxon/1001131

### ISSUE THREAD

**Issue by jhpoelen** [mammaldiversity/mammaldiversity.github.io#64]
to reproduce:

1. visit https://www.mammaldiversity.org/taxon/1001131
2. locate document link
3. click on http://digitallibrary.amnh.org/bitstream/handle/2246/582//v2/dspace/ingest/pdfSource/bul/B003a08.pdf?sequence=1&isAllowed=y

![image](https://github.com/mammaldiversity/mammaldiversity.github.io/assets/1084872/2438703f-3511-41a5-93b8-6836526ce4f6)

**Comment by JelleZijlstra** [comment]
This is fixed in the unreleased version, we now link to http://hdl.handle.net/2246/582 for this species.

**Comment by jhpoelen** [comment]
Suggest to include the md5 hash of the linked document.

**Comment by jhpoelen** [comment]
there appears to be a bunch more non-200 OK responses associated with the authority links.

```
curl "https://www.mammaldiversity.org/mdd.json" | jq --raw-output .[].authoritySpeciesLink | grep -v null | xargs -L1 curl -I 
```

**Comment by jhpoelen** [comment]
Also, did you consider building a corpus of literature similar to https://bat-literature.github.io / or https://taxodros.github.io . These pesky download links are likely to do dead sooner or later. . .

**Comment by JelleZijlstra** [comment]
I have virtually all confirmed references as PDFs. I'd like to publish those at some point but haven't pursued it, partly because we'd have to deal with copyright restrictions.

We're aiming to link species descriptions to the Biodiversity Heritage Library where possible, so we don't rely on too many small websites that may all change or go down. This one either isn't on BHL or my script couldn't find it.

The unreleased version mostly replaces the existing links in MDD with links from my Hesperomys database. I found a lot of dead links while doing that replacement, but I haven't done a full scan like you suggested above. I might add a script that crawls all the links every so often (once a year?) and flags ones that no longer work.

**Comment by n8upham** [comment]
Awesome, great to have those links resolving via [http://hdl.handle.net/] -- is that your own service for Hesperomys, @JelleZijlstra ?

**Comment by JelleZijlstra** [comment]
No, HDL is a general document-resolving service. I don't know too much about it but it's apparently run by CNRI: https://www.handle.net/. I try to use HDL links (or DOIs) if possible because they are likely to be more stable than the raw URLs.
